public/scripts/maps.js

```diff
@@ -23,31 +24,77 @@ $(document).ready(function() {
   const mobile = $(window).width() <= 500;
   renderSVG(mobile, $('#floor').select2('data')[0].text, true);
 
-  $('#toggle-2').change(function () {
+  //
+  $('#showBeacons').change(function () {
     if ($(this).is(':checked')) {
-      renderMockBeacons(mobile);
+      renderBeacons(mobile);
     } else {
       d3.selectAll('circle').remove();
     }
   });
 
-  $('#addmode').change(function () {
+  $('#showGateways').change(function () {
     if ($(this).is(':checked')) {
+      console.log('Showing gateways');
+    } else {
+      d3.selectAll('circle').remove();
+    }
+  });
+
+  $('#addmode').change(function () {
+    $('#addgateway').prop('checked', false);
```

Comment:
This doesn't really do anything for us. Visually the toggle is unchecked but no code is run to actually remove the handlers for that button